### PR TITLE
feat: transaction

### PR DESCRIPTION
**Pull request by rkdud007**
Implemented transaction both online mode & zkvm mode

### CODE REVIEW

**Comment by Okm165** [comment]
Arch refactor error handling improvements, SSZ merged!

**Review by Okm165 — APPROVED** [review]
LFG LGTM

File: lib/src/mpt.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    pub root: B256,
+}
+
+impl Mpt {
+    pub fn verify(&self, tx_index: u64, proof: Vec<Bytes>) -> Result<(), ProofVerificationError> {
+        let tx_encoded = alloy_rlp::encode(U256::from(tx_index));
+        let key = Bytes::from(tx_encoded);
+        let nibbles = Nibbles::unpack(key);
+
+        // TODO: last element of proof is the value of the key, not sure if it's ok to hardcode split prefix
```
Comment (rkdud007) [inline lib/src/mpt.rs:17]: bump on this

**Comment by rkdud007** [comment]
lets do A. value parse from proof B. step optimization in next PR -- it gets huge

**Pull request merged**